### ISSUE TITLE
Wavelets---Selecting Denoise disables contrast levels above level 6

### ISSUE THREAD

**Issue by RichardRegal** [Beep6581/RawTherapee#5938]
Version: 5.8-2482-gea156df1a
Branch: dev
Windows 10 version 2004

It is possible to have up to 10 levels (9+Extra) in the contrast part of wavelets. However, if denoise and refine in wavelets is selected then levels 7 and above are disabled even though the program allows one to make adjustments to those levels. This is to save memory as denoise uses a lot of memory. This causes 2 problems.

(_a_) If I have selected 9 levels and denoise then the residual image adjustments work only on the residual after the 10th level even though contrast adjustments work only on levels 1-6. If I have selected 6 levels then the residual image adjustments work on the residual after the 6th level.

(_b_) I use the higher levels to provide shape to the image and this is usually in levels 6–8. This means I have to choose between providing shape or reducing the noise. Obviously I would like to do both.

I think either there should be an option to use both and take the performance hit or selecting denoise should toggle off somehow the higher wavelet levels and should redefine the residual image to be the residual after what can be adjusted. In any event there should be better description in the program labels and sliders.

See https://discuss.pixls.us/t/are-wavelets-working-as-designed/20624/5 for discussion.

**Comment by Desmis** [comment]
@RichardRegal 

I just open a new branch "waveltimp" where I chnage the GUI

If level > 7 - denoise cannot be selected, and is disable
if level = 7 and denoise enable - chroma fine is enable

I think It is a simple solution, that solve many cases

jacques

**Comment by RichardRegal** [comment]
I shall have to wait until it appears in the automated build list. In the meantime I have a couple of questions.

(_a_) Suppose I start with levels at 6 then I enable denoise and make some adjustments. If I then go back and select 9 levels I assume my denoise adjustments are ignored. Does the GUI now show denoise as being disabled? I think there should be text in the tool tip that mentions selecting levels above 6 disables the denoise. (And possibly on the denoise tool tip indicating that enabling it disables the contrast levels above 6.)

(_b_) If I select 7 levels and denoise do the contrast adjustments still work on the 7th level or is it just the chroma adjustments? The reason I ask is that the contrast currently does not work in the 7th level if denoise is selected.

**Comment by Desmis** [comment]
@RichardRegal 
Yes your denoise works are ignored

Now all works with level 7

But you have 2 alternatives for denoise
* use denoise in "detail tab"
* use denoise in "Local adjustments" - 
** this module is more efficient than that of wavelet. But actually it is not completely  usefull, because in full image it uses a big lot of memory @heckflosse , Ingo is working on. But in most cases "full image" is not necessary. 
**In this last update when Ingo will have finish, you will have 2 majors improvments - denoise using local contrast mask, and equalizer hue

jacques

**Comment by RichardRegal** [comment]
@Desmis 
Thanks. With contrast going up to level 7 I can probably do most of the shaping* I want. I don’t often use level 8 and have never used level 9.

I must admit it had not occurred to me to use the local adjustments (and having looked at it I probably still wouldn’t). I do use the denoise in the detail tab but sometimes wavelets denoise does the job so much better (see Andy Astbury’s video at https://www.youtube.com/watch?v=j2cVxO2bGMo).

*This is what I mean by shaping. I zoom right out until the photo is maybe 6cm wide on the screen (about 5%). That way I can see what the picture is about and see the major outlines. Then I go to full screen view and select the wavelet level that gives me those outlines clearly but just those outlines and bump up the contrast at that (or those) levels. Usually it is in levels 6–8.

The details tab also has a contrast by levels tool, but it is not quite so easy to isolate where that would have an effect or understand what the levels are.

**Comment by Desmis** [comment]
@RichardRegal 

Can I merge these changes ?

jacques

**Comment by RichardRegal** [comment]
@Desmis 
Yes, please merge the changes. They certainly give me more than I have now and make the program easier to use.

Of course, I always reserve the right to ask for more :-)

**Comment by Desmis** [comment]
@RichardRegal 

I just merge

Jacques
